### DISCUSSION TITLE
Are the $focus examples broken? Is $focus broken?

### DISCUSSION THREAD

**Discussion by richcarni** [alpinejs/alpine#4043] [7. Bugs (archived)]
I've tried in 3 browsers across 2 oses, but the - click a button and use left and right arrow - $focus examples aren't working for me https://alpinejs.dev/plugins/focus

Is it just me?

I can't get it to work in my code either, although able to workaround using something like this: `$nextTick(() => { $refs.button.focus()`

**Comment by ekwoka** [comment]
there's something GOOFY with the CSS.

The `focus:ring-2` isn't working.

The focus itself works though, I think...

Your issue with $nextTick is likely just that the ref hasn't been detected yet.

**Reply by joshhanley** [reply]
Yeah it looks like the CSS hasn't been compiled. If you remove the Tailwind ring classes, you can see the focus state.

Edit: I've messaged Caleb to get him to check the CSS build.

**Comment by richcarni** [comment]
Thanks for looking into this - turns out the website issues are separate to the problem I was experiencing...

Which, I'm embarrassed to say, was nothing more than  user error - I'm sure you would have picked it up in 2 second if I'd included a code snippet of what wan't working - I'd simply neglected the leading `$focus` in `$focus.focus(el)`

**Reply by ekwoka** [reply]
interesting that that did not throw a more explicit error that would have helped track it...
